### ISSUE TITLE
Templated pages do not build for "off page" links on DataTable

### ISSUE THREAD

**Issue by archiewood** [evidence-dev/evidence#1115]
## Steps To Reproduce

1. Create a DataTable which contains links to a templated page per row
2. Create > 10 rows, but only display 10 on the first page
3. Deploy the site (or `npm run build` and inspect the assets created)
4. Attempt to navigate to one of the templated pages that is not shown on the first page of table results on the deployed site

## Expected Behavior

The templated page is loaded

## Actual Behaviour

There is no page, you get a 404

<img width="328" alt="CleanShot 2023-08-17 at 16 33 48@2x" src="https://github.com/evidence-dev/evidence/assets/58074498/944e7eee-db1d-4b62-b2e3-d5674d6c52bf">

https://github.com/evidence-dev/evidence/assets/58074498/c12c4271-b286-43e7-b0c7-4b87f029a0f3

## Why does this happen

1. Evidence renders a site statically: It has to know the links for all the pages in advance.
1. Evidence crawls the site, and finds all the links to other pages.
   1. Any pages it finds links for, it builds
1. When a table is included, only the rows in the first page of the table are included in the static render
   1. The other rows in the table are rendered dynamically using js when the user goes to later pages of the table.
1. So when evidence does the crawl it only finds the links that are visible in page one, and only these are built

## Workarounds

You need to make sure that every page you want to render has a link somewhere in Evidence that will be picked up at build time. Some options:

1. Extend the number of displayed rows of your table so that all rows are visible
1. Add at least one link to each templated page elsewhere on the site eg
    1. using a list of all the pages somewhere
    1. Or perhaps by putting a link to "next item" on all items, so that the pages self reference to each other. 

[Related Slack Thread](https://evidencedev.slack.com/archives/C023LRB9Z40/p1692302096895539)

**Comment by archiewood** [comment]
Somewhat related: https://github.com/evidence-dev/evidence/issues/1046

**Comment by csjh** [comment]
Could also do something like:
```html
    {#if link}
        <div style="display: none">
            {#each Array.from(new Set(data.map(row => row[link]))) as unique_link}
                <a href={unique_link}>a</a>
            {/each}
        </div>
    {/if}
```
Inside `DataTable.svelte`

**Comment by archiewood** [comment]
Interesting.

A nice thing about a more general solution is that I suspect there may be other components that have this behaviour too.

What if the only link to a page is inside a modal, for example? Would that render?

**Comment by csjh** [comment]
I believe it's just any links that are present when SSR happens, so no for the modal

**Comment by mcrascal** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Evidence-642db831f82932001aa2b3d1/issues/evidence-dev/evidence/1115?planning-poker) with Zenhub @csjh @ItsMeBrianD

**Comment by csjh** [comment]
Relevant SK source:

```js
for (const href of hrefs) {
    if (!is_root_relative(href)) continue;

    const { pathname, search, hash } = new URL(href, 'http://localhost');

    if (search) {
        // TODO warn that query strings have no effect on statically-exported pages
    }

    if (hash) {
        const key = decode_uri(pathname + hash);

        if (!expected_hashlinks.has(key)) {
            expected_hashlinks.set(key, new Set());
        }

        /** @type {Set<string>} */ (expected_hashlinks.get(key)).add(decoded);
    }

    enqueue(decoded, decode_uri(pathname), pathname);
}
```
(context: `enqueue` queues a page to be parsed for `hrefs` to run through this same loop)

So yeah no way to pass page params or hash through to tell the DataTable to cycle through the pages

**Comment by csjh** [comment]
Fixed by #1132
